pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -1266,6 +1267,12 @@ private SchemaStorage createAndStartSchemaStorage() throws Exception {
         return schemaStorage;
     }
 
+    private void setSchemaRegistryName(SchemaStorage schemaStorage) {
+        if (schemaStorage == null) {
+            this.config.setSchemaRegistryClassName("org.apache.pulsar.broker.service.schema.DefaultSchemaRegistryService");
+        }
+    }
```

Comment:
What is the purpose of this method? It's not immediately clear why we need it.

Reply:
in the existing implementation there's a null check on schema storage. if schemaStorage is null then use DefaultSchemaRegistryService as the schema registry, otherwise use SchemaRegistryServiceImpl. this method is performing the same check.

Reply:
Hmmm, looks like `DefaultSchemaRegistryService` is not a reasonable name, it should be `SchemaRegistryServiceDisabled`.